src/bacnet/getevent.c

```diff
@@ -18,7 +18,7 @@
  * @param lastReceivedObjectIdentifier  Object identifier
  * @return Bytes encoded.
  */
-int getevent_apdu_encode(uint8_t *apdu,
+static int getevent_apdu_encode(uint8_t *apdu,
```

Comment:
can getevent_apdu_encode() be added to header file instead?

Reply:
Of course. You know the best what should be moved to headers. Just say and I will move those ones.

Reply:
I notice that in header we already have getevent_encode_apdu() so now we would have

```c
    BACNET_STACK_EXPORT
    int getevent_apdu_encode(
        uint8_t *apdu,
        BACNET_OBJECT_ID *lastReceivedObjectIdentifier);

    BACNET_STACK_EXPORT
    int getevent_encode_apdu(
        uint8_t * apdu,
        uint8_t invoke_id,
        BACNET_OBJECT_ID * lastReceivedObjectIdentifier);
```

This looks strange as those are almost identical. But I do not feel yet that I can make call here. So please advice is this ok or not. I can change it anyway if needed.

Reply:
tl;dr; add DEPRECATED() to getevent_encode_apdu() declaration, include both declarations in header file

The original naming and features of the 'encode' of a service added the invoke ID instead of just initializing the APDU header separately.  I found when working in another project with this entanglement, that it is better to have separate service encode and service header encode.

The naming convention that is easiest for me to maintain is module_name+noun+verb, or even bacnet+module_name+noun+verb to help give a unique namespace.  The bacnet+module_name+noun+verb - for encoders and decoders would include apdu and apdu_size.  The shorter version, more for existing code, includes just apdu without size and can still be used, first with a call where apdu=NULL to get the size, and if size is okay, then call with apdu=buffer to fill the buffer.

Reply:
Can you please review that last commit do what you wanted.